build/build.go

```diff
@@ -520,6 +520,10 @@ func toSolveOpt(ctx context.Context, d driver.Driver, multiDriver bool, opt Opti
 		so.FrontendAttrs["platform"] = strings.Join(pp, ",")
 	}
 
+	if len(opt.Platforms) > 1 && opt.Exports[0].Type == "docker" {
```

Comment:
Looks correct but move it to line 458 so all the docker exporter handling is in one place.